packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -569,12 +581,25 @@ export class ConnectionManager implements IConnectionManager {
 				lastError = origError;
 
 				const retryDelayFromError = getRetryDelayFromError(origError);
-				delayMs = retryDelayFromError ?? Math.min(delayMs * 2, MaxReconnectDelayInMs);
-
 				if (retryDelayFromError !== undefined) {
+					// If the error told us to wait, then we wait.
 					this.props.reconnectionDelayHandler(retryDelayFromError, origError);
+					await new Promise<void>((resolve) => {
+						setTimeout(resolve, retryDelayFromError);
+					});
+				} else if (globalThis.navigator?.onLine !== false) {
```

Comment:
It feels like we can remove the check for globalThis.navigator?.onLine here and make it unconditional.
There is small change in behavior, but it feels like it does not matter, and code becomes simpler / smaller.

Reply:
That was my initial implementation but it introduces the performance regression you called out in your original comments.  I do think skipping the delay when transitioning to online makes sense to get a faster connection.